### ISSUE TITLE
下载这个链接 必现 失败

### ISSUE THREAD

**Issue by hnn6326** [AriaLyy/Aria#968]
## 版本
* 框架版本

* 系统版本

## 错误的url
https://oss.kaoyanvip.cn/material/E7ozbjqfePmLsdZ2YBGkkM.jpg

## 错误日志

![e227d6480957802d96abb1533269924](https://user-images.githubusercontent.com/26315467/181458050-1c4cf6b9-d9fa-41f3-8853-f621958f2ce5.jpg)

## 重现步骤

1. 默认下载配置
2.开始下载 上面链接必现

### 以下是我的猜测：
![1d9edf06dc23cc531ef7050037e7fac](https://user-images.githubusercontent.com/26315467/181462994-ea1c2649-8e50-46af-8580-bb482d5a95d5.jpg)
![bdc45c9417b0661ae61ba9c0f354c8a](https://user-images.githubusercontent.com/26315467/181463021-740d758a-1183-4df4-a73e-01bf6506a50c.jpg)

未分段单线程下载 走到上面图中逻辑， 写文件后没有flush 然后通知进度， merge的时候是从本地读文件大小 是0 导致的merge失败， 然后又自动删除了。     所以报失败而且 异常信息为null

**Comment by hemantp66** [comment]
facing the same issue in the below URL 
https://proxy-21.sg1.dailymotion.com/sec(De5fxjsSMLluDNsywYrHM_xxGWbAKWXcbj3P1HKkQwiy1M7lhvVT-HHKnJuAI0EP0E1RvZ36ogSsAXQCXKAHuQ)/video/420/733/505337024_mp4_h264_aac_fhd.mp4

got any solution ?

**Comment by hnn6326** [comment]
> facing the same issue in the below URL https://proxy-21.sg1.dailymotion.com/sec(De5fxjsSMLluDNsywYrHM_xxGWbAKWXcbj3P1HKkQwiy1M7lhvVT-HHKnJuAI0EP0E1RvZ36ogSsAXQCXKAHuQ)/video/420/733/505337024_mp4_h264_aac_fhd.mp4
> 
> got any solution ?

Yes , you can manual dependency PublicComponent module .  The author has dealt with this problem in the master branch of the PublicComponent library .

1、  Manually eliminate the original  public library。
  api(rootProject.ext.dependencies["aria"]) {
        exclude group: 'me.laoyuyu.aria', module: 'public'
    }
2、download the  PublicComponent  to you project and apply

**Comment by hemantp66** [comment]
Thanks for Help, 
i applied PublicComponent this way
`
 implementation("me.laoyuyu.aria:core:3.8.16")
            {
                exclude group: 'me.laoyuyu.aria', module: 'public'
            }
    implementation project(path: ':PublicComponent')
    
    kapt "me.laoyuyu.aria:compiler:3.8.16"
`

But this is not working for me
